### ISSUE TITLE
[Parent] nucleus-docs: add a11y documentation to each component

### ISSUE THREAD

**Issue by evecrabb** [centrica-engineering/nucleus-docs#816]
## Outcome

According to our [strategy](https://centricaplc-my.sharepoint.com/:p:/r/personal/rob_tobias_centrica_com/Documents/Microsoft%20Teams%20Chat%20Files/v1-4%20-%20DS%20and%20XEng%20Strategy%20H2%20-%202025%203.pptx?d=wdba2f2455d30449fb3290e88a2f0e5bc&csf=1&web=1&e=UUGkKC&nav=eyJzSWQiOjI5NSwiY0lkIjoxMTgzNzk3NDQyfQ) we want a11y sections within the documentation for each component. 
This parent will help to manage and define the work required to get us there. I'd recommend this is done iteratively rather than epic-based as this could be a sizeable chunk of documentation. 

Each component should have information pertaining to: 
- Interactions
  - with a mouse
  - with a keyboard
  - perhaps, maybe even with a screenreader but this changes device-to-device. 
  - hover/focus styles
- developer considerations
  - aria-labels, if required?
  - tabIndex information, if required

### Example of MD we could use: 

```md
## Accessibility
### Interactions
#### Mouse
#### Keyboard

### Developer considerations
#### aria-labels
list out the implications
#### tabIndex

```

### a11y fails are more likely to be caught

By documenting what a component should be in order to achieve compliance, we identify any gaps.

### Trust is increased in the team

If stakeholders know that we've covered all the bases, confidence in our components and systems are increased. 

### Education and awareness increase

If all of our components have an a11y documentation piece, we transfer more knowledge to stakeholders about best practices, keeping a11y top of mind across the business. 

## Scope

For each sub-task (component defined as needing a11y docs):

- [ ] craft an accessibility docs section

### Refinement

- [ ] Clear outcome
- [ ] Team understands
- [ ] PO understands

### Discovery

- [ ] Designs/Assets ready for development
- [ ] Draft documentation
- [ ] Shown to team & refine rules
- [ ] PO/Clockwork Approval
- [ ] Refined scope inside GitHub card

### Final Review

- [ ] Documentation
- [ ] Code
- [ ] Accessibility checks
- [ ] Integration with AEM6
